File: carddrawer/src/test/java/com/meli/android/carddrawer/configuration/CardFontConfigurationFactoryTest.java
package com.meli.android.carddrawer.configuration;

import com.meli.android.carddrawer.BasicRobolectricTest;

import org.junit.Test;
import org.junit.runner.RunWith;
import org.robolectric.RobolectricTestRunner;

import static org.junit.Assert.assertTrue;

@RunWith(RobolectricTestRunner.class)
public class CardFontConfigurationFactoryTest extends BasicRobolectricTest {

    @Test
    public void getConfiguration_withDarkType_returnsDarkFontConfiguration() {
        CardFontConfiguration configuration = CardFontConfigurationFactory.getConfiguration(FontType.DARK_TYPE, 0, getContext());

        assertTrue(configuration instanceof DarkFontConfiguration);
    }

    @Test
    public void getConfiguration_withLightType_returnsLightFontConfiguration() {
        CardFontConfiguration configuration = CardFontConfigurationFactory.getConfiguration(FontType.LIGHT_TYPE, 0, getContext());

        assertTrue(configuration instanceof LightFontConfiguration);
    }

    @Test
    public void getConfiguration_withInvalidType_returnsDefaultFontConfiguration() {
        CardFontConfiguration configuration = CardFontConfigurationFactory.getConfiguration("invalid_type", 2, getContext());

        assertTrue(configuration instanceof DefaultFontConfiguration);
    }

}

File: carddrawer/src/test/java/com/meli/android/carddrawer/configuration/CardFontConfigurationFactoryTest.kt
package com.meli.android.carddrawer.configuration

import com.meli.android.carddrawer.BaseTest
import org.junit.Test

class CardFontConfigurationFactoryTest : BaseTest() {

    @Test
    fun `when call function getConfiguration of CardFontConfigurationFactory with FontType DARK_TYPE per parameter then return DarkFontConfiguration`() {
        val configuration = CardFontConfigurationFactory.getConfiguration(FontType.DARK_TYPE, 0, contextMock)
        assert(configuration is DarkFontConfiguration)
    }

    @Test
    fun `when call function getConfiguration of CardFontConfigurationFactory with FontType LIGHT_TYPE per parameter then return LightFontConfiguration`() {
        val configuration = CardFontConfigurationFactory.getConfiguration(FontType.LIGHT_TYPE, 0, contextMock)
        assert(configuration is LightFontConfiguration)
    }

    @Test
    fun `when call function getConfiguration of CardFontConfigurationFactory with FontType DARK_NO_SHADOW_TYPE per parameter then return DarkFontConfiguration`() {
        val configuration = CardFontConfigurationFactory.getConfiguration(FontType.DARK_NO_SHADOW_TYPE, 0, contextMock)
        assert(configuration is DarkFontConfiguration)
    }

    @Test
    fun `when call function getConfiguration of CardFontConfigurationFactory with FontType LIGHT_NO_SHADOW_TYPE per parameter then return LightFontConfiguration`() {
        val configuration = CardFontConfigurationFactory.getConfiguration(FontType.LIGHT_NO_SHADOW_TYPE, 0, contextMock)
        assert(configuration is LightFontConfiguration)
    }

    @Test
    fun `when call function getConfiguration of CardFontConfigurationFactory with FontType NONE per parameter then return DefaultFontConfiguration`() {
        val configuration = CardFontConfigurationFactory.getConfiguration(FontType.NONE, 0, contextMock)
        assert(configuration is DefaultFontConfiguration)
    }

}